File: tarea-2/Program/Inventario.java
package Program;

import java.util.ArrayList;
import java.util.Comparator;

import Program.Items.Item;

public class Inventario {
    private Mascota mascota;
    private ArrayList<Item> inventario = new ArrayList<Item>();

    public Inventario(Mascota mascota){
        this.mascota = mascota;
    }
    
    public ArrayList<Item> abrirInventario(){
        return inventario;
    }
    
    public void agregarItem(Item item){
        inventario.add(item);
    }
    
    // retorna verdadero si se cantidad = 0, falso si no
    public boolean consumirItem(Item item){
        item.usarItem(mascota);
        if(item.getCantidad()==0){
            inventario.remove(item); // JVM sabrá cual item eliminar porque difieren en ID
            return true;
        }
        return false;
    }

    public Item buscarItem(int ID){
        Item aEnonctrar = null;
        for (Item i: inventario){
            if (i.getId() == ID){
                aEnonctrar = i;
                break;
            }
        }
        return aEnonctrar;
    }

    public void sort(){
        inventario.sort(new Comparator<Item>() {
            @Override
            public int compare(Item i1, Item i2){
                return Integer.compare(i1.getId(), i2.getId());
            }
        });
    }

}

File: tarea-3/etapa-4/src/Inventario.cpp
#include <iostream>
#include <string>
#include <vector>
#include "../headers/Item.h"
#include "../headers/Inventario.h"

Inventario::Inventario(Mascota &mascota) : mascota(mascota) 
{}

Inventario::~Inventario() {
    for (Item *item : items) {
		delete item;
	}
}

void Inventario::agregarItem(Item *item) {
    items.push_back(item);
}

Item * Inventario::buscarPorID(int id) {
	for (Item *item : items) {
		if (item->getId() == id)
		    return item;
	}
	return NULL;
}

void Inventario::usarItem(int id, Mascota &mascota) {
	Item *item = buscarPorID(id);
    item->usar(mascota);

	int index = 0;
	for (Item *item : items) {
		if (item->getId() == id)
            break;
		index++;
	}

	if (item->getCantidad() == 0) {
		items.erase(items.begin() + index);
	}
}

void Inventario::mostrarInventario() const{
	for (Item *item : items) {
		item->mostrarItem();
	}
};
